docs/03_tutorials/20_tic-tac-toe-game-smart-contract-single node.md

```diff
@@ -0,0 +1,761 @@
+
+### Procedure to create the tictactoe.cpp file
+Complete the following steps to create the tictactoe.cpp file:
+
+1. Ensure you are in the tictactoe folder. 
+2. Create the tictactoe.cpp file.
+3. Open the tictactoe.cpp file with your text editor.
+4. Import the tictactoe.hpp file and make the tictactoe definitions from the previous section available.
+   Add this code to the .cpp file: 
+
+```c++
+#include "tictactoe.hpp"
+```
+5. Implement Create.
+   * Ensure that the action has the signature from the host
+   * Ensure that the challenger and host are not the same player
+   * Ensure that there is no existing game
+   * Store the newly created game to the multi index table
+
+   Add this code to the .cpp file:
+
+```c++
+void tictactoe::create(const name &challenger, name &host) {
+    require_auth(host);
+    check(challenger != host, "Challenger should not be the same as the host.");
+
+    // Check if game already exists
+    games existingHostGames(get_self(), host.value);
+    auto itr = existingHostGames.find(challenger.value);
+    check(itr == existingHostGames.end(), "Game already exists.");
+
+    existingHostGames.emplace(host, [&](auto &g) {
+        g.challenger = challenger;
+        g.host = host;
+        g.turn = host;
+    });
+}
+```
+
+6. Implement Restart.
+   * Ensure that the action has the signature from the host/challenger
+   * Ensure that the game exists
+   * Ensure that the restart action is done by host/challenger
+   * Reset the game
+   * Store the updated game to the multi index table
+
+   Add this code to the .cpp file:
+
+```c++
+void tictactoe::restart(const name &challenger, const name &host, const name &by){
+    check(has_auth(by), "Only " + by.to_string() + "can restart the game.");
+
+    // Check if game exists
+    games existingHostGames(get_self(), host.value);
+    auto itr = existingHostGames.find(challenger.value);
+    check(itr != existingHostGames.end(), "Game does not exist.");
+
+    // Check if this game belongs to the action sender
+    check(by == itr->host || by == itr->challenger, "This is not your game.");
+
+    // Reset game
+    existingHostGames.modify(itr, itr->host, [](auto &g) {
+        g.resetGame();
+    });
+}
+```
+
+
+7. Implement Close.
+   * Ensure that the action has the signature from the host
+   * Ensure that the game exists
+   * Remove the game from the db
+
+   Add this code to the .cpp file:
+
+```c++
+void tictactoe::close(const name &challenger, const name &host){
+    check(has_auth(host), "Only the host can close the game.");
+
+    require_auth(host);
+
+    // Check if game exists
+    games existingHostGames(get_self(), host.value);
+    auto itr = existingHostGames.find(challenger.value);
+    check(itr != existingHostGames.end(), "Game does not exist.");
+
+    // Remove game
+    existingHostGames.erase(itr);
+}
+```
+
+8. Implement Move supporting methods.
+   * Implement IsValidMove.
+   Add this code to the .cpp file:
```

Comment:
\`isValidMove()\` (wrapping backticks and trailing parentheses on function names).
Also, add extra line between these two lines, or else it renders in one line.